pandas/core/reshape/melt.py

```diff
@@ -70,7 +72,13 @@ def melt(frame, id_vars=None, value_vars=None, var_name=None,
 
     mdata = {}
     for col in id_vars:
-        mdata[col] = np.tile(frame.pop(col).values, K)
+        id_data = frame.pop(col)
+        if is_extension_type(id_data):
+            # Preserve pandas dtype by not converting to a numpy array
+            id_data = concat([id_data] * K, ignore_index=True)
```

Comment:
@TomAugspurger do we have this in ExtensionArray ATM? ``.tile()``? or could emulate like this

Reply:
No `.tile`. We do have `_concat_same_type`.